### ISSUE TITLE
How to use Notes in an Action, not in a Question

### ISSUE THREAD

**Issue by san-ouadghiri** [serenity-js/serenity-js#225]
Hi,

New day, new question!

On a website, I have an element, with the href attribute. I want to store the href value (so, a URL). Later on, down the scenario, I want to browse to that URL. By searching how to do that, here's where I'm stuck.

-> Seek an attribute value: Attribute.of(MySite.consultButton).called('href')
-> Store the variable: TakeNote seems to be the right method for me here.

**So, cast/actor side:**
```
import { protractor } from 'protractor';
import { Actor, TakeNotes } from 'serenity-js/lib/screenplay';
import { BrowseTheWeb, Cast } from 'serenity-js/protractor';

export class Persona implements Cast {

    actor(firstNameActor: string): Actor {

                return Actor.named(firstNameActor)
                    .whoCan(BrowseTheWeb.using(protractor.browser))
                    .whoCan(TakeNotes.usingAnEmptyNotepad())
                    ;
        }
    }
}
```
**Scenario side:**
```
it("Offer (1/3)", () => san.attemptsTo(
   Open.browserOn(HOME_PAGE),
   RetrieveDomain.using(),
   //Some other stuff
   //I'd like here to browse to the retrieved domain
)
```
I initially tried `BrowseTheWeb.as(san).locate(MySite.consultButton).getAttribute('href'),` but then I realize TakeNote was needed.

**Task side:**
export class RetrieveDomain implements Task {

    public static using(): RetrieveDomain {
        return new RetrieveDomain();
    }

    performAs(actor: PerformsTasks): PromiseLike<void> {
        //const SiteUrl = Attribute.of(MySite.consultButton).called('href').toString();
        return actor.attemptsTo(TakeNote.of(SiteUrl))
    }
}

**Interaction side:**
I'm actually not sure where to declare SiteUrl. So I tried as an interaction. I tried both
```
export class SiteUrl {
    static Url = Attribute.of(MySite.consultButton).called('href')
}
```
or
`export const SiteUrl = { Url: Attribute.of(MySite.consultButton).called('href')}`

Both case I got, task side (on retrieveDomain.ts):

> Error:(17, 45) TS2345: Argument of type 'typeof SiteUrl' is not assignable to parameter of type 'Question<{}>'. Property 'answeredBy' is missing in type 'typeof SiteUrl'.

I also tried to log this and got
```
AttributeValue {
[e2e]   target: 
[e2e]    Target {
[e2e]      name: 'The Button',
[e2e]      locator: name(name) {
[e2e]     return By.css('*[name="' + escapeCss(name) + '"]');
[e2e]   } { using: 'xpath', value: '//a[@data-custom="the-btn"]' } },
[e2e]   attribute: 'href',
[e2e]   toString: [Function] }
```
So... not sure if that's the log.console that is not helpful, or if Attribute.of is not doing what I think it does :(.

**So to sum up:**
- Where and how should I declare & retrieve my attribute value.
- How can I use this, not in a question, but in an action (browse.To(Site.Url) kind of line).
- Bonus : How can I declare and use a task that has no arguments? Each time I try not using a decorator (like .using()) I got an error, so I keep using an empty useless one.

**Comment by jan-molak** [comment]
Hi @san-ouadghiri and apologies for not being able to get round to this ticket sooner!
I was focusing on Serenity/JS 2.0. The good news is that it's now live and ships with [better docs around taking notes](https://serenity-js.org/modules/core/class/src/screenplay/interactions/TakeNote.ts~TakeNote.html)

Please check out the [API docs](https://serenity-js.org/modules/core/class/src/screenplay/interactions/TakeNote.ts~TakeNote.html) and [unit tests](https://serenity-js.org/modules/core/test-file/spec/screenplay/interactions/TakeNote.spec.ts.html) that demonstrate usage and let me know what you think!

Best,
Jan
